planner/core/testdata/integration_suite_out.json

```diff
@@ -881,5 +881,88 @@
         ]
       }
     ]
+  },
+  {
+    "Name": "TestIndexJoinOnClusteredIndex",
+    "Cases": [
+      {
+        "SQL": "select /*+ inl_join(t1, t2) */ * from t t1 join t t2 on t1.a = t2.a",
+        "Plan": [
+          "IndexJoin_10 3.00 root  inner join, inner:TableReader_7, outer key:test.t.a, inner key:test.t.a",
+          "├─TableReader_25(Build) 3.00 root  data:TableFullScan_24",
+          "│ └─TableFullScan_24 3.00 cop[tikv] table:t1 keep order:false",
+          "└─TableReader_7(Probe) 1.00 root  data:TableRangeScan_6",
+          "  └─TableRangeScan_6 1.00 cop[tikv] table:t2 range: decided by [test.t.a], keep order:false"
+        ],
+        "Res": [
+          "1 111 1.1000000000 11 1 111 1.1000000000 11",
+          "2 222 2.2000000000 12 2 222 2.2000000000 12",
+          "3 333 3.3000000000 13 3 333 3.3000000000 13"
+        ]
+      },
+      {
+        "SQL": "select /*+ inl_merge_join(t1, t2) */ * from t t1 join t t2 on t1.a = t2.a",
+        "Plan": [
+          "IndexMergeJoin_12 3.00 root  inner join, inner:TableReader_9, outer key:test.t.a, inner key:test.t.a",
+          "├─TableReader_25(Build) 3.00 root  data:TableFullScan_24",
+          "│ └─TableFullScan_24 3.00 cop[tikv] table:t1 keep order:false",
+          "└─TableReader_9(Probe) 1.00 root  data:TableRangeScan_8",
+          "  └─TableRangeScan_8 1.00 cop[tikv] table:t2 range: decided by [test.t.a], keep order:true"
```

Comment:
why `keep order` is true here?

Reply:
Because it's the inner side of `IndexMergeJoin`.